buildSrc/src/main/rocker/com/uber/okbuck/template/config/OkbuckAndroidModules.rocker.raw

```diff
@@ -0,0 +1,113 @@
+        if (':src_' in dep and not dep.endswith(':src_main')):
+            mapped_res_deps.append(src_to_res(dep))
+    return mapped_res_deps
+
+# Macro that represents a union of all common rules to for an android_library
+def okbuck_android_module(
+        name,
+        language = None,
+        ## common args
+        visibility = None,
+        package=None,
+        ## Manifest related args
+        manifest_main_manifest = None,
+        manifest_min_sdk = None,
+        manifest_target_sdk = None,
+        manifest_version_code=None,
+        manifest_version_name=None,
+        manifest_debuggable=False,
+        manifest_secondary_manifests=None,
+        ## Resources related args
+        res_res = None,
+        res_project_res = None,
+        res_assets = None,
+        res_resource_union=None,
+        res_extra_deps = [],
+        ## Android library related args
+        deps = [],
+        exported_deps = [],
+        **kwargs
+    ):
+
+    manifest_rule_name = name.replace("src", "manifest_lib", 1)
+    okbuck_manifest(
+        name = manifest_rule_name,
+        main_manifest = manifest_main_manifest,
+        min_sdk = manifest_min_sdk,
+        target_sdk = manifest_target_sdk,
+        package = package,
+        version_code = manifest_version_code,
+        version_name = manifest_version_name,
+        debuggable = manifest_debuggable,
+        secondary_manifests = manifest_secondary_manifests,
+    )
+
+    res_ext_aar_deps = get_exts_aar_deps(deps)
+    res_lib_deps = get_libs_res_deps(deps)
+    res_deps = res_ext_aar_deps + res_lib_deps + res_extra_deps
+
+    res_exported_ext_aar_deps = get_exts_aar_deps(exported_deps)
+    res_exported_lib_deps = get_libs_res_deps(exported_deps)
+    res_exported_deps = res_exported_ext_aar_deps + res_exported_lib_deps
+
+    res_rule_name = name.replace("src", "res", 1)
+    res_kwargs = dict(
+        name = res_rule_name,
+        deps = res_deps if res_deps else None,
+        res = res_res,
+        assets = res_assets,
+        project_res = res_project_res,
+        package = package,
+        resource_union = res_resource_union,
+        visibility = visibility
+    )
+    if res_deps:
+      res_kwargs.update({'deps' : res_deps})
+    if res_exported_deps:
+      res_kwargs.update({'exported_deps' : res_exported_deps})
+
+    @(androidResourceRule)(**res_kwargs)
+
+    src_kwargs = dict(
+        name = name,
+        visibility = visibility,
+        manifest = get_rule_dep(manifest_rule_name),
+    )
+    src_kwargs.update(kwargs)
+
+    src_deps = deps + res_lib_deps + [get_rule_dep(res_rule_name)]
+    src_exported_deps = exported_deps + res_exported_lib_deps
+
+    src_kwargs.update({'deps' : src_deps})
+    if src_exported_deps:
+        src_kwargs.update({'exported_deps' : src_exported_deps})
+    if language:
+        src_kwargs.update({'language' : language})
+
+    @(androidLibraryRule)(**src_kwargs)
+
+def okbuck_kotlin_android_module(
```

Comment:
we should allow overriding `okbuck_kotlin_android_module` & `okbuck_android_module` as well.

This way we can override them to be `uber_kotlin_android_module` & `uber_android_module` in the android repo. These override can then directly call respective `okbuck_*` impl.

Reply:
This is already allowed.
Although it's kinda of strange to declare, since you'd have to call the `nativeRuleName` as `android_module`.
I'll update the PR with the example so people can refer to